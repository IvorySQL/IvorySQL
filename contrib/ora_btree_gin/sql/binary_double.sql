-- binary_double check

CREATE TABLE binary_double_test (a binary_double);

\copy binary_double_test from 'data/binary_double.data'

SET enable_seqscan=on;

SELECT count(*) FROM binary_double_test WHERE a <  -1890.0;

SELECT count(*) FROM binary_double_test WHERE a <= -1890.0;

SELECT count(*) FROM binary_double_test WHERE a  = -1890.0;

SELECT count(*) FROM binary_double_test WHERE a >= -1890.0;

SELECT count(*) FROM binary_double_test WHERE a >  -1890.0;

CREATE INDEX binary_doubleidx ON binary_double_test USING gin ( a );

SET enable_seqscan=off;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_double_test WHERE a <  -1890.0::binary_double;
SELECT count(*) FROM binary_double_test WHERE a <  -1890.0::binary_double;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_double_test WHERE a <= -1890.0::binary_double;
SELECT count(*) FROM binary_double_test WHERE a <= -1890.0::binary_double;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_double_test WHERE a  = -1890.0::binary_double;
SELECT count(*) FROM binary_double_test WHERE a  = -1890.0::binary_double;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_double_test WHERE a >= -1890.0::binary_double;
SELECT count(*) FROM binary_double_test WHERE a >= -1890.0::binary_double;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_double_test WHERE a >  -1890.0::binary_double;
SELECT count(*) FROM binary_double_test WHERE a >  -1890.0::binary_double;
