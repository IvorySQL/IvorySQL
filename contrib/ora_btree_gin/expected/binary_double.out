-- binary_double check
CREATE TABLE binary_double_test (a binary_double);
\copy binary_double_test from 'data/binary_double.data'
SET enable_seqscan=on;
SELECT count(*) FROM binary_double_test WHERE a <  -1890.0;
 count 
-------
   237
(1 row)

SELECT count(*) FROM binary_double_test WHERE a <= -1890.0;
 count 
-------
   238
(1 row)

SELECT count(*) FROM binary_double_test WHERE a  = -1890.0;
 count 
-------
     1
(1 row)

SELECT count(*) FROM binary_double_test WHERE a >= -1890.0;
 count 
-------
   307
(1 row)

SELECT count(*) FROM binary_double_test WHERE a >  -1890.0;
 count 
-------
   306
(1 row)

CREATE INDEX binary_doubleidx ON binary_double_test USING gin ( a );
SET enable_seqscan=off;
EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_double_test WHERE a <  -1890.0::binary_double;
                       QUERY PLAN                       
--------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on binary_double_test
         Recheck Cond: (a < '-1890'::binary_double)
         ->  Bitmap Index Scan on binary_doubleidx
               Index Cond: (a < '-1890'::binary_double)
(5 rows)

SELECT count(*) FROM binary_double_test WHERE a <  -1890.0::binary_double;
 count 
-------
   237
(1 row)

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_double_test WHERE a <= -1890.0::binary_double;
                       QUERY PLAN                        
---------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on binary_double_test
         Recheck Cond: (a <= '-1890'::binary_double)
         ->  Bitmap Index Scan on binary_doubleidx
               Index Cond: (a <= '-1890'::binary_double)
(5 rows)

SELECT count(*) FROM binary_double_test WHERE a <= -1890.0::binary_double;
 count 
-------
   238
(1 row)

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_double_test WHERE a  = -1890.0::binary_double;
                       QUERY PLAN                       
--------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on binary_double_test
         Recheck Cond: (a = '-1890'::binary_double)
         ->  Bitmap Index Scan on binary_doubleidx
               Index Cond: (a = '-1890'::binary_double)
(5 rows)

SELECT count(*) FROM binary_double_test WHERE a  = -1890.0::binary_double;
 count 
-------
     1
(1 row)

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_double_test WHERE a >= -1890.0::binary_double;
                       QUERY PLAN                        
---------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on binary_double_test
         Recheck Cond: (a >= '-1890'::binary_double)
         ->  Bitmap Index Scan on binary_doubleidx
               Index Cond: (a >= '-1890'::binary_double)
(5 rows)

SELECT count(*) FROM binary_double_test WHERE a >= -1890.0::binary_double;
 count 
-------
   307
(1 row)

EXPLAIN (COSTS OFF)
SELECT count(*) FROM binary_double_test WHERE a >  -1890.0::binary_double;
                       QUERY PLAN                       
--------------------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on binary_double_test
         Recheck Cond: (a > '-1890'::binary_double)
         ->  Bitmap Index Scan on binary_doubleidx
               Index Cond: (a > '-1890'::binary_double)
(5 rows)

SELECT count(*) FROM binary_double_test WHERE a >  -1890.0::binary_double;
 count 
-------
   306
(1 row)

